File: Team5/RotateArray.cpp
class Solution {
public:
    void reverse(vector<int>& v, int begin, int end){
        while(begin < end){
            int temp = v[begin];
            v[begin] = v[end];
            v[end] = temp;
            begin++;
            end--;
        }
    }
    void rotate(vector<int>& nums, int k) {
        k = k%nums.size();
        // REVERSE THE WHOLE ARRAY
        reverse(nums, 0, nums.size()-1);
        // REVERSE THE ARRAY TILL K;
        reverse(nums, 0, k-1);
        // REVERSE THE ARRAY FROM K+1 TO THE END
        reverse(nums, k, nums.size()-1);
    }
};

File: Team1/RiyaSingh/rotateArray.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        int j=0;
        if(k>n){
            k = k%n;
        }
        int temp[] = new int[n];
        for(int i=0;i<k;i++){
            temp[i] = nums[n-k+i];
        }
        for(int i=k;i<n;i++){
            temp[i] = nums[j];
            j++;
        }
        System.arraycopy(temp,0,nums,0,n);
    }
}
